src/qiboml/models/decoding.py

```diff
@@ -84,19 +85,30 @@ def __call__(
         Returns:
             (CircuitResult | QuantumState | MeasurementOutcomes): the execution ``qibo.result`` object.
         """
+        x = self.preprocessing(x)
+
+        return self.backend.execute_circuit(x + self._circuit, nshots=self.nshots)
+
+    def preprocessing(self, x: Circuit) -> Circuit:
```

Comment:
```suggestion
    def _preprocessing(self, x: Circuit) -> Circuit:
```

If we don't plan to expose it.
If you agree with this, we also have to update the name everywhere it is used.

Reply:
OK, fine! Then I think we just need to add a oneline docstring here as well :)